T/t8code/build_tarballs.jl

```diff
@@ -0,0 +1,51 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg
+
+name = "t8code"
+version = v"1.1.0"
+
+# Collection of sources required to complete build
+sources = [
+    ArchiveSource("https://github.com/DLR-AMR/t8code/releases/download/v$(version)/t8code_v$(version).tar.gz", "01ceae6dd0ed596cf1abb5bc78caa77b5da94ca60d080aae3ccf0ba3c6464deb")
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir
+cd t8code/
+./configure --enable-mpi CFLAGS=-O3 CXXFLAGS=-O3 CC=mpicc CXX=mpicxx --prefix=${prefix} --build=${MACHTYPE} --host=${target}
+make -j
+make install
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = [
+    Platform("i686", "linux"; libc = "glibc"),
+    Platform("x86_64", "linux"; libc = "glibc"),
+    Platform("aarch64", "linux"; libc = "glibc"),
+    Platform("armv6l", "linux"; call_abi = "eabihf", libc = "glibc"),
+    Platform("armv7l", "linux"; call_abi = "eabihf", libc = "glibc"),
+    Platform("powerpc64le", "linux"; libc = "glibc")
+]
+
+
+# The products that we will ensure are always built
+products = [
+    LibraryProduct("libt8", :libt8),
+    LibraryProduct("libsc", :libsc),
+    LibraryProduct("libp4est", :libp4est)
+]
+
+# Dependencies that must be installed before this package can be built
+dependencies = [
+    Dependency(PackageSpec(name="MPICH_jll", uuid="7cb0a576-ebde-5e09-9194-50597f1243b4"))
+    Dependency(PackageSpec(name="MPItrampoline_jll", uuid="f1f71cc9-e9ae-5b93-9b94-4fe0e1ad3748"))
```

Comment:
This is slighlty over-optimistic. See L/LAMMPS for an example of how to use the MPI platform

Reply:
> What do you mean by "over-optimistic"?

Smashing all MPI implementations together is bound to end up very bad: you're just linking to a single implementation, the only one surviving this: https://buildkite.com/julialang/yggdrasil/builds/1409#0186e07f-9078-4309-aa3a-1c1b7fce553c/637-1751.  Which one is it it's everybody's guess.  What Valentin suggested is a more straightforward way to deal with the different implementations.

Reply:
I left out `MPItrampoline_jll` for now.